internal/services/cdn/cdn_frontdoor_security_policy_resource.go

```diff
@@ -272,19 +288,114 @@ func resourceCdnFrontdoorSecurityPolicyDelete(d *pluginsdk.ResourceData, meta in
+	policyType := input[0].(map[string]interface{})
+	firewallPolicy := policyType["firewall"].([]interface{})
+	v := firewallPolicy[0].(map[string]interface{})
+
+	if id := v["cdn_frontdoor_firewall_policy_id"].(string); id != "" {
+		results.WafPolicy = &securitypolicies.ResourceReference{
+			Id: utils.String(id),
+		}
+	}
+
+	configAssociations := v["association"].([]interface{})
+
+	for _, item := range configAssociations {
+		v := item.(map[string]interface{})
+		domains := expandSecurityPoliciesActivatedResourceReference(v["domain"].([]interface{}))
+
+		if isStandardSku {
+			if len(*domains) > 100 {
+				return &results, fmt.Errorf("the 'Standard_AzureFrontDoor' sku is only allowed to have 100 or less domains associated with the firewall policy, got %d", len(*domains))
+			}
+		} else {
+			if len(*domains) > 500 {
+				return &results, fmt.Errorf("the 'Premium_AzureFrontDoor' sku is only allowed to have 500 or less domains associated with the firewall policy, got %d", len(*domains))
+			}
+		}
+
+		association := securitypolicies.SecurityPolicyWebApplicationFirewallAssociation{
+			Domains:         domains,
+			PatternsToMatch: utils.ExpandStringSlice(v["patterns_to_match"].([]interface{})),
+		}
+
+		associations = append(associations, association)
+	}
+
+	results.Associations = &associations
+
+	return &results, nil
+}
+
+func expandSecurityPoliciesActivatedResourceReference(input []interface{}) *[]securitypolicies.ActivatedResourceReference {
+	results := make([]securitypolicies.ActivatedResourceReference, 0)
+	if len(input) == 0 {
+		return &results
+	}
+
+	for _, item := range input {
+		v := item.(map[string]interface{})
+
+		if id := v["cdn_frontdoor_domain_id"].(string); id != "" {
+			results = append(results, securitypolicies.ActivatedResourceReference{
+				Id: utils.String(id),
+			})
+		}
+	}
+
+	return &results
+}
+
+func flattenSecurityPoliciesActivatedResourceReference(input *[]securitypolicies.ActivatedResourceReference) ([]interface{}, error) {
+	results := make([]interface{}, 0)
+	if input == nil {
+		return results, nil
+	}
+
+	for _, item := range *input {
+		frontDoorDomainId := ""
+		if item.Id != nil {
+			if parsedFrontDoorCustomDomainId, frontDoorCustomDomainIdErr := parse.FrontDoorCustomDomainIDInsensitively(*item.Id); frontDoorCustomDomainIdErr == nil {
+				frontDoorDomainId = parsedFrontDoorCustomDomainId.ID()
+			} else if parsedFrontDoorEndpointId, frontDoorEndpointIdErr := parse.FrontDoorEndpointIDInsensitively(*item.Id); frontDoorEndpointIdErr == nil {
+				frontDoorDomainId = parsedFrontDoorEndpointId.ID()
+			} else {
+				return nil, fmt.Errorf("flattening `cdn_frontdoor_domain_id`: %+v; %+v", frontDoorCustomDomainIdErr, frontDoorEndpointIdErr)
+			}
+		}
+
+		active := false
+		if item.IsActive != nil {
+			active = *item.IsActive
+		}
+
+		results = append(results, map[string]interface{}{
+			"active":                  active,
+			"cdn_frontdoor_domain_id": frontDoorDomainId,
+		})
```

Comment:
I think we can simplify this to
```suggestion
		results = append(results, map[string]interface{}{
			"active":                  pointer.From(item.IsActive),
			"cdn_frontdoor_domain_id": frontDoorDomainId,
		})
```

Reply:
Fixed.